Missing swDest Could Use a Better Error message

**Library Affected**:
*workbox-build*

**Issue or Feature Request Description**:
Currently, if I list only a path for the `swDest` parameter of `generateSW()` I get the following error message:

```[ERROR] This happened: Error: Unable to write the service worker file.. 'EISDIR: illegal operation on a directory, open 'build''```

Something like this would be more helpful:

```swDest must contain both a valid path and valid file name.```

(While you're in there, you might see if you can remove the extra period from the end of 'Unable to write the service worker file.')

--- a/packages/workbox-build/src/lib/errors.js
+++ b/packages/workbox-build/src/lib/errors.js
@@ -25,6 +25,8 @@ module.exports = {
   'read-sw-template-failure': 'Unable to read the service worker template ' +
     'file.',
   'sw-write-failure': 'Unable to write the service worker file.',
+  'sw-write-failure-directory': 'Unable to write the service worker file; ' +
+    'swDest should be a full path to the file, not a path to a directory.',
   'unable-to-copy-workbox-sw': 'workbox-sw is needed by the service worker ' +
     'and could not be copied over to your new site.',
   'invalid-generate-sw-input': 'The input to generateSW() must be an ' +

--- a/packages/workbox-build/src/lib/write-sw.js
+++ b/packages/workbox-build/src/lib/write-sw.js
@@ -6,10 +6,10 @@ const runtimeCachingConverter = require('./utils/runtime-caching-converter');
 const template = require('lodash.template');
 
 module.exports =
-  (swSrc, manifestEntries, workboxSWImportPath, globDirectory, options) => {
+  (swDest, manifestEntries, workboxSWImportPath, globDirectory, options) => {
   options = options || {};
   try {
-    mkdirp.sync(path.dirname(swSrc));
+    mkdirp.sync(path.dirname(swDest));
   } catch (err) {
     return Promise.reject(
       new Error(`${errors['unable-to-make-sw-directory']}. '${err.message}'`)
@@ -80,11 +80,14 @@ module.exports =
   })
   .then((populatedTemplate) => {
     return new Promise((resolve, reject) => {
-      fs.writeFile(swSrc, populatedTemplate, (err) => {
-        if (err) {
-          return reject(
-            new Error(`${errors['sw-write-failure']}. '${err.message}'`)
-          );
+      fs.writeFile(swDest, populatedTemplate, (error) => {
+        if (error) {
+          if (error.code === 'EISDIR') {
+            // See https://github.com/GoogleChrome/workbox/issues/612
+            return reject(new Error(errors['sw-write-failure-directory']));
+          }
+          return reject(new Error(
+            `${errors['sw-write-failure']}. '${error.message}'`));
         }
 
         resolve();

--- a/packages/workbox-build/test/node/lib-write-sw.js
+++ b/packages/workbox-build/test/node/lib-write-sw.js
@@ -1,7 +1,6 @@
 const proxyquire = require('proxyquire');
 const errors = require('../../src/lib/errors.js');
-
-require('chai').should();
+const expect = require('chai').expect;
 
 describe('lib/write-sw.js', function() {
   const INJECTED_ERROR = new Error('Injected Error');
@@ -153,6 +152,29 @@ describe('lib/write-sw.js', function() {
         throw new Error('Unexpected error thrown: ' + caughtError.message);
       }
     });
+  });
+
+  it('should handle error writing file due to swDest being a directory', async function() {
+    const writeSw = proxyquire('../../src/lib/write-sw', {
+      'mkdirp': {sync: () => {}},
+      'fs': {
+        readFile: (pathname, encoding, cb) => {
+          cb(null, 'Injected Template');
+        },
+        writeFile: (filepath, stringToWrite, cb) => {
+          cb({code: 'EISDIR'});
+        },
+      },
+      'lodash.template': () => () => 'Injected populated template.',
+    });
+
+    try {
+      await writeSw('fake-path/', [{url: '/', revision: '1234'}],
+        'workbox-sw.min.js', 'fake-path/');
+      throw new Error('Expected error to be thrown');
+    } catch(error) {
+      expect(error.message).to.eql(errors['sw-write-failure-directory']);
+    }
   });
 
   it('should be able to generate sw for template', function() {

